### PR TITLE
doc: add traceback trick and beam shift VA convention

### PR DESCRIPTION
**Pull request by pieleric**

### CODE REVIEW

File: doc/develop/extending.rst [hunk 1]
```diff
@@ -539,9 +539,30 @@ to observe the internal state of the python processes and place breakpoints.
...
+It's possible to display the complete exception this way:
+
+.. code-block:: python
+
+   try:
+       ccd.data.get()  # Failing code
+   except Exception as ex:
+       try:
+           remote_tb = ex._pyroTraceback
+           rmt_ex = "Remote exception %s" % ("".join(remote_tb),)
```
Comment (nandishjpatel) [inline doc/develop/extending.rst:555]:
```suggestion
           rmt_ex = f'Remote exception {"".join(remote_tb)}'
```
Just a suggestion, not necessary to implement

**Review by nandishjpatel — APPROVED** [review]
Looks good to me!

File: doc/develop/backend-api.rst [hunk 2]
```diff
@@ -237,20 +242,46 @@ a single component. That is why Scanner inherit from Emitter.
...
 
     .. Rationale: we could have done slightly differently by using a general .transformation (VA, array of float, shape of (3,3) for a 2D resolution). It would have been a transformation matrix from the scanning area to the actual value. Very generic, but more complex to use and read and the advanced transformations possible don't seem to be useful.
 
+    .. py:attribute:: shift
+
+        *(VA, tuple of floats, unit=m, optional)* How much shift is applied to the center of the area acquired.
+        It is expressed in meters.
+        A bigger value in X means a shift towards the left (ie, the center of the new image corresponds to a point
+        at the left of the center of the previous image).
+        A bigger value in Y means a shift towards the bottom (ie, the center of the new image corresponds to a point
```
Comment (K4rishma) [inline doc/develop/backend-api.rst:276]: According to me, when the shift value of the e-beam in Y is positive, it moves upwards and the image seems to move downwards.  I am not sure if it tries to convey this or something else.

File: doc/develop/backend-api.rst [hunk 3]
```diff
@@ -237,20 +242,46 @@ a single component. That is why Scanner inherit from Emitter.
...
 
         *(VA, float, unit=rad)* counter-clockwise rotation to apply on the original area to obtain the actual area to scan.
 
     .. Rationale: we could have done slightly differently by using a general .transformation (VA, array of float, shape of (3,3) for a 2D resolution). It would have been a transformation matrix from the scanning area to the actual value. Very generic, but more complex to use and read and the advanced transformations possible don't seem to be useful.
 
+    .. py:attribute:: shift
+
+        *(VA, tuple of floats, unit=m, optional)* How much shift is applied to the center of the area acquired.
+        It is expressed in meters.
+        A bigger value in X means a shift towards the left (ie, the center of the new image corresponds to a point
```
Comment (K4rishma) [inline doc/develop/backend-api.rst:274]: I am not sure what is meant be this line in the bracket (e, the center of the new image corresponds to a point)

File: doc/develop/backend-api.rst [hunk 4]
```diff
@@ -237,20 +242,46 @@ a single component. That is why Scanner inherit from Emitter.
...
 
-    .. py:attribute:: scale
+    .. py:attribute:: translation
 
-        *(VA, tuple of floats or int, unit=ratio)* ratio of the size of the scannable area divided by the size of the scanned area. Note that this is the inverse of the typical definition of scale (i.e., increasing the scale leads to a smaller scanned area). The advantage of this definition is that its meaning is very similar to binning. Note that the MD_PIXEL_SIZE metadata of a dataflow will depend both on pixelSize and scale (i.e., MD_PIXEL_SIZE = pixelSize * scale).
+        *(VA, tuple of floats, unit=px)* How much shift is applied to the center of the area acquired.
+        It is expressed in pixels (the size of a pixel being defined by pixelSize, and so independent of .scale).
+        It is limited by the field of view of the scanner, and it is automatically
+        clipped based on the current resolution and scale.
+        A bigger value in X means a shift towards the right, and a bigger value in Y means a shift towards the bottom.
```
Comment (K4rishma) [inline doc/develop/backend-api.rst:259]: For the image perspective this is correct. Physically the beam moves in opposite directions. Maybe be something to clear?

File: doc/develop/backend-api.rst [hunk 5]
```diff
@@ -237,20 +242,46 @@ a single component. That is why Scanner inherit from Emitter.
...
 
     .. py:attribute:: rotation
 
         *(VA, float, unit=rad)* counter-clockwise rotation to apply on the original area to obtain the actual area to scan.
 
     .. Rationale: we could have done slightly differently by using a general .transformation (VA, array of float, shape of (3,3) for a 2D resolution). It would have been a transformation matrix from the scanning area to the actual value. Very generic, but more complex to use and read and the advanced transformations possible don't seem to be useful.
 
+    .. py:attribute:: shift
+
+        *(VA, tuple of floats, unit=m, optional)* How much shift is applied to the center of the area acquired.
```
Comment (K4rishma) [inline doc/develop/backend-api.rst:272]: Maybe mentioning the order of tuple i.e first X and then Y would be helpful?

**Review by K4rishma — COMMENTED** [review]
I have added comments based on the findings of our pair-programming on the beam shift and others are suggestions that might improve the readability a bit.

File: doc/develop/backend-api.rst (hunk 3 above)
Comment (pieleric) [reply doc/develop/backend-api.rst:274]: point -> position. "the center of the new image corresponds to a position at the left of the center of the previous image". Is that clear enough? Or how would you say it?

File: doc/develop/backend-api.rst (hunk 2 above)
Comment (pieleric) [reply doc/develop/backend-api.rst:276]: That's the case in the SimSEM, but not with the xt_client, which does the opposite (and in X too). We'll adjust the SimSEM (and orsay driver) later.

File: doc/develop/backend-api.rst (hunk 3 above)
Comment (K4rishma) [reply doc/develop/backend-api.rst]: It is clear for me. I did not read the next line as I thought it ended at point.

**Review by K4rishma — APPROVED** [review]

**Pull request merged**